dbs/meilisearch/scripts/bulk_index.py

```diff
@@ -161,13 +167,17 @@ async def main(files: list[str]) -> None:
             _update_sortable_attributes(client, "wines"),
         )
         index = client.index("wines")
-        tasks = []
         print("Processing files")
-        for file in files:
-            data = read_jsonl_from_file(file)
-            data = validate(data, Wine, exclude_none=True)
-            tasks.append(do_indexing(index, data, file))
-            print(f"Validated data from {Path(file).name} in pydantic")
+        with ProcessPoolExecutor() as process_pool:
+            loop = asyncio.get_running_loop()
+            calls = [partial(process_file, file_name) for file_name in files]
+            call_coroutines = []
+
+            for call in calls:
+                call_coroutines.append(loop.run_in_executor(process_pool, call))
+
+            data = await asyncio.gather(*call_coroutines)
```

Comment:
Wow, this is a really neat trick! Just so I understand, are you creating process pool coroutines and attaching them as tasks within an async event loop? If so, is the whole thing (CPU-bound pydantic and I/O-bound file and db ops) running in a **single** event loop? How is the whole thing happening in a non-blocking fashion?

In any case, very cool, need to try this out in other dbs as well. Thanks for this PR!

Reply:
I was messing around with `aiofiles` in this case and it was turning out to be slower than the regular sequential file processing. I presume that's because there's some CPU overhead due to the data validation in Pydantic, and that the bottleneck in the file-processing in this case is not purely I/O-bound? Curious to hear your thoughts on this too.

I know I'm working with a bit of a toy example here, but in a real-world situation, it's very likely I'll be using Pydantic extensively to handle issues with data and ensure the right data is being indexed. I'm eager to see what Pydantic 2.x (in Rust) brings to this fold, looking forward to running some experiments on that!

Reply:
Great, thanks for clarifying, and again, for the PR! 😄